lib/concurrent/edge/lock_free_linked_set/node.rb

```diff
@@ -0,0 +1,72 @@
+require 'concurrent/edge/atomic_markable_reference'
+
+module Concurrent
+  module Edge
+    class LockFreeLinkedSet
+      class Node < Synchronization::Object
+        include Comparable
+
+        attr_reader :Data, :SuccessorReference, :Key
```

Comment:
This does not feel right `pred.SuccessorReference.compare_and_set ...`. Could you create underscore readers for the CalmelCase ivars?